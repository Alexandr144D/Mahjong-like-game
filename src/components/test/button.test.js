import React from 'react';
import sinon from 'sinon'
import {shallow, mount, render} from 'enzyme';
import ButtonComponent from '../Button';


describe("Button component", () => {

    it("Button toMatchSnapShot", () => {
        const button = shallow(<ButtonComponent />);

        expect(button).toMatchSnapshot();
    });

    it("Button toBeDefined", () => {
        const button = shallow(<ButtonComponent />);

        expect(button).toBeDefined()
    });

    it("Button name 'Start Game'", () => {
        const button = shallow(<ButtonComponent />);

        expect(button.text()).toBe("Start Game")
    });

    it("Button name 'Start Game'", () => {
        const button = shallow(<ButtonComponent />);

        const clickCallback = sinon.spy();
        button.setProps({startGame: clickCallback});
        button.find('.makeStyles-button-1').simulate("click");

        sinon.assert.called(clickCallback);
    });
});
