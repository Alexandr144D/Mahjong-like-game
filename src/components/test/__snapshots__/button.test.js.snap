// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button toMatchSnapShot 1`] = `ShallowWrapper {}`;
